src/main/scala/wdl4s/cwl/CwlType.scala

```diff
@@ -0,0 +1,51 @@
+package wdl4s.cwl
+
+import eu.timepit.refined._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.string._
+import mouse.boolean._
+import shapeless.{:+:, CNil}
+
+object CwlType extends Enumeration {
+  type CwlType = Value
+
+  val Null = Value("null")
+  val Boolean = Value("boolean")
+  val Int = Value("int")
+  val Long = Value("long")
+  val Float = Value("float")
+  val Double = Value("double")
+  val String = Value("string")
+  val File = Value("File")
+  val Directory = Value("Directory")
+}
+
+case class File(
+  `class`: String Refined MatchesRegex[W.`"File"`.T],
+  location: Option[String], //TODO refine w/ regex  of IRI
+  path: Option[String],
+  basename: Option[String],
+  dirname: Option[String],
+  nameroot: Option[String],
+  nameext: Option[String],
+  checksum: Option[String],
+  size: Option[Long],
+  secondaryFiles: Array[File :+: Directory :+: CNil],
+  format: Option[String],
+  contents: Option[String])
+
+object File {
+  def validateContents: File => Either[String, Unit] =  //this is a sample validation.  There are about a billion of them, not sure
+    f =>
+      (!(f.location.isEmpty && f.path.isEmpty && f.contents.isEmpty)).
```

Comment:
the validation of Files is [painful](http://www.commonwl.org/v1.0/Workflow.html#File)  I thought I could get it started but I am going to create a ticket as it is pretty tricky and this is nowhere near complete.  

In fact I'm just going to delete this as it's the only piece of downstream validation there is.